### DISCUSSION TITLE
Add assigned_to username to label

### DISCUSSION THREAD

**Discussion by BertCurbit** [snipe/snipe-it#10228] [Q&A] [answered]
Hi

Is there a way to add the username (assigned to the asset) to the label? I've played around with the labls_blade.php file and added the assigned_to field but it returns the ID and not the username.

Thanks in advance

**Comment by MonkeyNPC** [comment, marked as answer]
This is how I have mine set up in */resources/views/hardware/labels.blade.php:

Under <div class="qr_text!='')

**add:** 

```
 @if (($asset->assigned_to!=' ') && ($asset->assigned_type=='App\Models\User'))

		<div class="pull-left">

		    Assigned To: {{ $asset->assignedto->first_name }} {{ $asset->assignedto->last_name }}

		</div>
@endif 

```
Where ever you would like the Assigned to user to appear.  It will write it as "First Name Last Name" on the label.
Example: "Assigned To: John Smith" You can change the "Assigned To:" at the beginning to be Checked Out To: or any other option including removing that part completely. You can also add html before and after such as
`
 <b><u>Assigned To:</u></b>
`

I'm trying to figure out from here how to add assignment options based on location/parent assets. If anyone knows how to do this, please reply.

**Reply by MonkeyNPC** [reply]
Just solved one of my own quesitons. To add "Assigned to: Location" if you check items out to location rather than a specific user. You can add this to your "labels.blade.php":
```

	    @if (($asset->assigned_to!=' ') && ($asset->assigned_type=='App\Models\Location'))
		<div class="pull-left">
		    <b><u>Assigned To</u></b>: {{ $asset->assignedto->name }}
		</div>
	    @endif

```

And finally, if you check out certain items such as a charger or keyboard to other devices you can add this to display the parent asset that the item is checked out to:
```

	    @if (($asset->assigned_to!=' ') && ($asset->assigned_type=='App\Models\Asset'))
		<div class="pull-left">
		    <b><u>Assigned To</u></b>: {{ $asset->assignedto->name }}
		</div>
	    @endif

```

^ If you would like to have it show the parent asset's asset tag instead of just the device name you can change **{{ $asset->assignedto->name }}** to **{{ $asset->assignedto->asset_tag }}** instead.

**Comment by brianhoganm** [comment]
I just came across this and just wanted to say thank you! Very helpful.
